src/lib/comms/sol-oic-security.c

```diff
@@ -53,34 +51,51 @@ struct cred_item {
     /* FIXME: Only symmetric pairwise keys supported at the moment. */
 };
 
-struct creds {
-    struct sol_vector items;
-    const struct sol_oic_security *security;
-};
-
 struct sol_socket *sol_coap_server_get_socket(const struct sol_coap_server *server);
 
 static ssize_t
 creds_get_psk(const void *data, struct sol_str_slice id,
     char *psk, size_t psk_len)
 {
-    const struct creds *creds = data;
-    struct cred_item *iter;
-    uint16_t idx;
+    SOL_BUFFER_DECLARE_STATIC(buf, PATH_MAX);
+    struct sol_cert *cert;
+    struct sol_blob *contents;
+    int r;
 
     SOL_DBG("Looking for PSK with ID=%.*s", (int)id.len, id.data);
 
-    SOL_VECTOR_FOREACH_IDX (&creds->items, iter, idx) {
-        if (sol_str_slice_eq(id, iter->id.slice)) {
-            if (iter->id.slice.len > psk_len)
-                return -ENOBUFS;
+    r = sol_buffer_append_printf(&buf, "%s", OIC_CRED_FILE_PREFIX);
+    SOL_INT_CHECK(r, < 0, -ENOBUFS);
+    r = sol_buffer_append_as_base64(&buf, id, SOL_BASE64_MAP);
```

Comment:
could you clarify why? I'm wondering if we should do an escape for filenames, like we did for http/url, this way most of the chars would be preserved and we have a faster process by doing memcpy() instead of the conversion to base64. Usually the convention is to do like sytemd-escape (https://www.freedesktop.org/software/systemd/man/systemd-escape.html). See https://github.com/solettaproject/soletta/blob/master/src/lib/comms/sol-http-common.c#L140, we could use a variation of that.